vllm/core/block/block_table.py

```diff
@@ -274,18 +297,30 @@ def _get_all_token_ids(self) -> List[int]:
         if not self._is_allocated:
             return token_ids
 
-        for block in self._blocks:
-            token_ids.extend(block.token_ids)
+        for block in self.blocks:
+            cur_token_ids = block.token_ids
+            if cur_token_ids is not None:
+                token_ids.extend(cur_token_ids)
 
         return token_ids
 
+    def _get_num_token_ids(self) -> int:
+        if not self._is_allocated:
+            return 0
```

Comment:
nit: I think we don't need this branch anymore. if it's not allocated, self.blocks will be empty

Reply:
Nice, removed